docs/source/index.rst

```diff
@@ -54,52 +54,33 @@ easy.
 **Step 1:** Pick the magic string that will identify your library. To
 avoid collisions, this should match your library's name on PyPI.
 
-**Step 2:** There's a special :class:`contextvars.ContextVar` object:
+**Step 2:** There's a special :class:`threading.local` object:
 
-.. data:: current_async_library_cvar
+.. data:: current_async_library_tlocal.name
 
-Make sure that whenever your library is running, this is set to your
-identifier string. In most cases, this will be as simple as:
+Make sure that whenever your library is calling a coroutine throw(), send(), or close()
+that this is set to your identifier string. In most cases, this will be as simple as:
 
 .. code-block:: python3
 
-   from sniffio import current_async_library_cvar
+   from sniffio import current_async_library_tlocal
 
-   # Your library's run function
-   def run(...):
-        token = current_async_library_cvar.set("my-library's-PyPI-name")
+   # Your library's step function
+   def step(...):
+        old_name, current_async_library_tlocal.name = (
```

Comment:
Would a context manager be a guardrail that would cause trouble?  Or perhaps just a performance difference?

```python
from sniffio import enter_async_library

def step(...):
    with enter_async_library("mylib"):
        result = coro.send(None)
```

Reply:
Can you say a bit more about why twisted would have to count stack frames and how that relates to guardrails?

Reply:
https://gitter.im/python-trio/general?at=60479824b5131f4f28021225
```
Kyle Altendorf @altendky 10:45
@graingert:matrix.org since this is related and we're discussing here in gitter: https://github.com/python-trio/sniffio/pull/23#discussion_r590338307 would providing a context manager be problematic?

graingert @graingert:matrix.org [m] 10:46
it's called every step in every coro framework

Kyle Altendorf @altendky 10:46
right, so performance is the concern?

graingert @graingert:matrix.org [m] 10:46
yeah I think so
```

Probably worth a comment that we are explicitly encouraging not using a context manager.

Reply:
> Can you say a bit more about why twisted would have to count stack frames and how that relates to guardrails?

Twisted counts stack frames for various purposes including determining what frame defer.returnValue was called from